### ISSUE TITLE
Investigate using DeepSpeed ZeRO plugin across multiple nodes

### ISSUE THREAD

**Issue by jackapbutler** [OpenBioML/chemnlp#205]
We want to check whether the Hugging Face `Trainer` will synchronise it's DeepSpeed training across multiple nodes through `torch.distributed` in the same way as it handles distributed data parallel training across multiple nodes.

If we were to use ZeRO across multiple nodes, how much extra memory is sharded and how much slower is training?

**Comment by jackapbutler** [comment]
We've completed a [multi-node](https://www.notion.so/stable-society/Pythia-2-8B-FT-v1-multi-node-4-5e36b870a85948c6bb2fb66d976bc1f9) training run with DeepSpeed using the Hugging Face Trainer. This confirmed that the HF `Trainer` does not sharding across multiple nodes automatically but does allow you to perform synchronised data parallel training inter-node using the setup you define in your single node script.

Basically replicate anything that works in a single node instantly across multiple nodes in a data parallel fashion.

**Comment by jackapbutler** [comment]
We've completed a [multi-node](https://www.notion.so/stable-society/Pythia-2-8B-FT-v1-multi-node-4-5e36b870a85948c6bb2fb66d976bc1f9) training run with DeepSpeed using the Hugging Face Trainer. 

This confirmed that when using DeepSpeed + HF `Trainer` across multiple nodes vs a single node you see a decrease in VRAM usage for the multi-node setup. This implies that DeepSpeed is detecting multiple nodes and automatically sharding state across node boundaries. This enables scaling our model size to even larger numbers of parameters and makes it possible for us to extrapolate directly from NVidia's [original paper estimates](https://arxiv.org/pdf/1910.02054.pdf).
